### DISCUSSION TITLE
[Feature] : Weekly Contest

### DISCUSSION THREAD

**Discussion by vedant-jain03** [CircuitVerse/CircuitVerse#3228] [Ideas]
## Description
A **weekly contest** is a new feature plan of the CircuitVerse.

Weekly contests on CircuitVerse will be open for every user. Users can register in contests and can submit one of their creations or Project on the CircuitVerse platform. The top projects and project authors will be the winner. The judging criteria for projects will be the votes by other users. The more votes, the higher will be the priority.

- The contest can be scheduled automated using ActiveJobs implementation in Rails. Every contest will be scheduled on Sunday and will end on Saturday(6 days).
- Users can register for only one project in a contest.
- The winning project will get featured on the `Featured Circuits` page. We can have a **_Special tag_** for winning projects with a tag message as `Weekly Contest Winner #{contest_number}`.
  Special Tag UI:
  ![Screenshot from 2022-07-05 16-51-36](https://user-images.githubusercontent.com/76901313/177316574-f92d9210-3db4-49b5-86bf-6039eac0e2fa.png)
- We can add an attribute in the user’s profile of `Weekly Contest Won`. For every winning project, the project’s author will get an increment in this attribute.
- Some changes to avoid Plagiarism in the Contest:
  - For a span of ending time of the contest, a registered project will not have an option of _Copy or Forked_.
  - A project that a user wanted to register for a contest should be their own project, not a forked one(Slug).
  - The circuit or project the user wants to register should be public.
- [Sample Contest Page](https://vedant-jain03.github.io/CircuitVerse_Sample_pages/).

## Expected Pages:
- **Contest Dashboard** (For Admins to control the contest flow manually).
- **Contests** (List all the previous and current contests to the user).
- **Contest** (Page for the respective contest).
- **Project Submission Form** (To register projects in the Contest).

This discussion is open to everyone, kindly provide your views on it.

**Comment by vedant-jain03** [comment]
cc @nitin10s @satu0king @tachyons @pavanjoshi914 @gr455 @pavanjoshi914 @abstrekt @ayan-biswas0412

**Comment by gr455** [comment]
In my opinion. Let us still have the option to copy and fork, we will just not enter the forked project into the contest. At the end of the day, our intention behind the contest is not to ACTUALLY have a contest to see who's the best, but to get some cool circuits to feature on the front page of the website. Disabling the fork button is overkill for this, not to mention we are cutting down a good feature to fork.

Thoughts or comments on this?

**Reply by vedant-jain03** [reply]
Yaa, my intention was not to take the forked and copied projects for the contest as the one who has the actual creation should get the credit. 
If the user wants to register their circuit we can just render only his/her circuits/projects!

**Comment by Asrani-Aman** [comment]
Hello maintainers,

I'd like to propose the inclusion of practice questions related to digital simulation on our platform, along with solutions for these questions. This approach would be similar to what platforms like LeetCode or GeeksforGeeks offer.
